### PR TITLE
Add Makefile and version file for Concourse

### PR DESCRIPTION
**Pull request by caismith-ch**
Resolves: PCI-405

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+.PHONY: clean
+clean:
+	rm -f $(bin)
+	rm -f *.zip
+	rm -rf build-*
+
+.PHONY: package
+package:
+	$(eval tmpdir:=$(shell mktemp -d build-XXXXXXXXXX))
+	cp -r ./$(cb) $(tmpdir)/$(cb)
```
Comment (jsimons1989) [inline Makefile]: companybindex, config and run-elastic-search.sh don't exist on the master branch yet.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+.PHONY: build
+build: fmt
+	go build
+
+.PHONY: test
+test: test-unit test-integration
+
+.PHONY: test-unit
+test-unit: test-deps
```
Comment (jsimons1989) [inline Makefile]: There's no test-deps target

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	@set -a; go test $(TESTS) -run 'Unit'
```
Comment (jsimons1989) [inline Makefile]: `@set -a;` isn't needed

File: Makefile (hunk 1 above)
Comment (caismith-ch) [reply Makefile]: Shall I comment it out for now as we know it will be required very soon?

Comment (jsimons1989) [reply Makefile]: Commented out code is usually bad practise and can cause confusion whether it is needed or not. I would advise removing for now. It'll be in the commit history when you need it.

Comment (caismith-ch) [reply Makefile]: Rightio

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+clean:
+	rm -f $(bin)
+	rm -f *.zip
+	rm -rf build-*
+
+.PHONY: dist
+dist: clean build package
+
+.PHONY: xunit-tests
+xunit-tests: test-deps
```
Comment (jsimons1989) [inline Makefile]: test-deps doesn't exist

**Review by jsimons1989 — APPROVED** [review]

**Pull request merged**